src/cmdstan/command.hpp

```diff
@@ -594,6 +594,149 @@ int command(int argc, const char *argv[]) {
+    to unconstrained scale
+    */
+    std::vector<std::string> param_names;
+    std::vector<std::vector<size_t>> param_dimss;
+    stan::services::get_model_parameters(model, param_names, param_dimss);
+
+    /*
+      Will store both constrained and unconstrained parameter sets in single
+      nested array so that only a single loop is required for calculation
+      and printing
+    */
+    size_t num_par_sets = c_params_vec_size + u_params_vec_size;
+    std::vector<std::vector<double>> params_r_ind(num_par_sets);
+
+    /*
+      An Eigen::Map with an inner stride is needed for viewing all parameter
+      values for a given set as a single vector when multiple parameter sets
+      are passed, as the var_context returns them appended by parameter, rather
+      than by set, i.e,:
+        [[a1, b1],[a2,b2]] => [a1, a2, b1, b2]
+    */
+    using StrideT = Eigen::Stride<1, Eigen::Dynamic>;
+    for (size_t i = 0; i < c_params_vec_size; i++) {
+      Eigen::Map<Eigen::VectorXd, 0, StrideT> map_r(
+          c_params_r.data() + i, c_params_size, StrideT(1, c_params_vec_size));
+
+      // Constrained parameters are unconstrained directly into the std::vector
+      // of unconstrained parameter sets to iterate over
+      stan::io::array_var_context context(param_names, map_r, param_dimss);
+      model.transform_inits(context, dummy_params_i, params_r_ind[i], &msg);
+    }
+
+    for (size_t i = c_params_vec_size; i < num_par_sets; i++) {
+      size_t iter = i - c_params_vec_size;
+      Eigen::Map<Eigen::VectorXd, 0, StrideT> map_r(
+          u_params_r.data() + iter, u_params_size,
+          StrideT(1, u_params_vec_size));
+      // As they are already on the correct scale, the unconstrained parameters
+      // can simply be appended after the constrained (if any)
+      params_r_ind[i] = stan::math::to_array_1d(map_r);
+    }
+
+    std::string grad_output_file = get_arg_val<string_argument>(
+        parser, "output", "log_prob_output_file");
+    std::ofstream output_stream(grad_output_file);
+    output_stream << std::setprecision(sig_figs_arg->value()) << "lp_,";
+
+    std::vector<std::string> p_names;
+    model.constrained_param_names(p_names, false, false);
+    for (size_t i = 1; i < p_names.size(); i++) {
+      output_stream << "g_" << p_names[i] << ",";
+    }
+    output_stream << "g_" << p_names.back() << "\n";
+
+    double lp;
+    std::vector<double> gradients;
+    for (size_t i = 0; i < num_par_sets; i++) {
+      if (jacobian_adjust) {
+        lp = stan::model::log_prob_grad<false, true>(model, params_r_ind[i],
+                                                     dummy_params_i, gradients);
+      } else {
+        lp = stan::model::log_prob_grad<false, false>(
+            model, params_r_ind[i], dummy_params_i, gradients);
+      }
+
+      output_stream << lp << ",";
+
+      // Ensure that each row of values ends with a newline instead of delimeter
+      if (gradients.size() > 1) {
```

Comment:
Just use the general case for both.  I assume the `ostream_iterator` will do the right thing with sizes 0 and 1 and it's not worth making the code more complicated for something like this.  And then the `<< '\n'` can be on the same line (it can be a character instead of string).